code/statsd/src/main/scala/com/avast/datadog4s/StatsDEventFactory.scala

```diff
@@ -0,0 +1,23 @@
+package com.avast.datadog4s
+
+import cats.effect.{ Resource, Sync }
+import com.avast.datadog4s.api.EventFactory
+import com.timgroup.statsd.NonBlockingStatsDClientBuilder
+
+object StatsDEventFactory {
+  def make[F[_]: Sync](config: StatsDMetricFactoryConfig): Resource[F, EventFactory[F]] = {
+    val F = Sync[F]
+    Resource
+      .fromAutoCloseable(
+        F.delay(
+          new NonBlockingStatsDClientBuilder()
```

Comment:
i don't want to make this a blocker for this PR, but i think eventually we definitely want make it possible to share statsdclient between EventFactory and MetricsFactory

Reply:
I would rather solve it sooner than later.